File: leetcode/704._Binary_Search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        
        l, r = 0, len(nums)-1

        while l <= r:
            med = (l + r) // 2
            if nums[med] == target:
                return med
            elif nums[med] < target:
                l = med+1
            else: 
                r = med-1

        return -1

File: yandex_back_prep_cpp/704._Binary_Search.cpp
class Solution
{
public:
    int search(vector<int> &nums, int target)
    {
        int l = 0;
        int r = nums.size() - 1;

        while (l <= r)
        {
            int middle = (l + r) / 2;
            if (nums[middle] == target)
            {
                return middle;
            }
            else if (nums[middle] < target)
            {
                l = middle + 1;
            }
            else
            {
                r = middle - 1;
            }
        }

        return -1;
    }
};